### PR TITLE
libbot, yaml-cpp, and Eigen3 can be found using find_package()

### PR DESCRIPTION
**Pull request by fbudin69500**
* Supports CMake configuration files for libbot, yaml-cpp, and Eigen3
* Automatically finds glib-2.0 which is required by libbot

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -5,13 +5,38 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 
 add_executable(${APP_NAME} apriltags_driver.cpp LcmWrapper.cpp)
 
+find_package(libbot QUIET)
+if(NOT libbot_FOUND)
+  set(libbot_pkg bot2-core bot2-param-client lcmtypes_bot2-core)
+else()
+  # `lcmtypes/bot_core.hpp` is called `lcmtypes/bot_core.h` when CMake
```
Comment (patmarion) [inline src/CMakeLists.txt]: i thought `bot_core.h` was the C header and `bot_core.hpp` is the C++ header?

**Comment by patmarion** [comment]
Thanks for opening the PR, looks like a good improvement.  Can you explain the goal a little more?  With this change, it uses find_package, but falls back to pkg-config.  Why not always use one or the other?

File: src/apriltags_driver.cpp [hunk 2]
```diff
@@ -15,7 +15,11 @@
 #include <lcm/lcm-cpp.hpp>
 #include "LcmWrapper.hpp"
 
+#ifdef BOT_CORE_H
+#include <lcmtypes/bot_core.h>
+#else
```
Comment (patmarion) [inline src/apriltags_driver.cpp]: I don't understand this.  `bot_core` is widely used on many projects, why is this change necessary all of a sudden?  Is it possibly a bug in bot_core's cmake code?

**Comment by patmarion** [comment]
Hey, when you push new commits i don't get an alert so feel free to ping me if/when you want me to take another look.

So, I think I am happy to merge this as is but I was hoping to get clarification about the design before I do.  Please see my earlier question about why both find_package and pkg-config, why not one or the other?

**Comment by fbudin69500** [comment]
@patmarion I'll make sure to ping you when I push new commits. I was taking into account your comment and I am trying to update drake to simplify this PR. This is not quite ready to be merged. Thanks!

**Comment by fbudin69500** [comment]
@patmarion : This PR allows a user to use `find_package()` but keeps the previous compatibility with `pkg_config()`.

Drake generates `-config.cmake` files for libbot, yam-cpp, and Eigen. This PR allows apriltags_driver to consume these libraries compiled as part of drake (such as during Spartan's build) without breaking anything that is currently working. The current limitation is that libbot does not expose the include directories of its dependencies (i.e glib-2.0 and gthread-2.0) in its `-config.cmake` files, so the configuration still relies on `pkg_config()` to find these include directories. This is likey to be improved in the future.
